ecs-cli/modules/cli/local/up_app.go

```diff
@@ -46,26 +46,44 @@ func Up(c *cli.Context) {
 	if err := options.ValidateCombinations(c); err != nil {
 		logrus.Fatalf(err.Error())
 	}
-	composePath, err := createComposeFile(c)
+	composePath, err := createComposeProjectIfNotExist(c)
 	if err != nil {
-		logrus.Fatalf("Failed to get compose file path due to:\n%v", err)
+		logrus.Fatalf("Failed to create Compose file due to:\n%v", err)
 	}
+
 	runContainers(composePath)
 }
 
-// createComposeFile returns the path of the Compose file to start the containers from.
-// The Compose file ordering priority is defined as:
-// 1. Use the --task-def-compose flag if it exists
-// 2. Use the default docker-compose.local.yml file if no flags were provided and the file exists
-// 3. Otherwise, we need to create the Compose file.
-func createComposeFile(c *cli.Context) (string, error) {
-	if name := c.String(flags.TaskDefinitionCompose); name != "" {
-		return filepath.Abs(name)
+// createComposeProjectIfNotExist creates a new Compose file if necessary and returns its path.
+// If the user set the TaskDefinitionCompose flag, then return that Compose file path.
+// If the user doesn't have any flags set, and doesn't have LocalInFileName but has a LocalOutDefaultFileName,
+// then we use the LocalOutDefaultFileName file.
+// Otherwise, we create a new Compose file from the user's flags and return its path.
```

Comment:
Ah, also I see this now -- ignore previous review comment about adding comments ;)